### PR TITLE
ansible lint cleanup

### PR DESCRIPTION
**Pull request by gene-redpanda**
A bunch of changes to clean up ansible linter errors. Also ditched a few seriously questionable yamllint rules because they're straight up nonsense.

### CODE REVIEW

**Comment by gene-redpanda** [comment]
I set CI ready to only trigger on PR to main, that may be a mistake

File: ansible/roles/demo_certs/tasks/create-ca.yml [hunk 1]
```diff
@@ -3,46 +3,49 @@
...
-- name: create-ca - Initialize CA data files
+- name: Create-ca - initialize CA data files
   delegate_to: localhost
-  shell: |
+  ansible.builtin.shell: |
     cd {{ root_ca_dir }}
     rm -f index.txt serial.txt
     touch index.txt
     echo '01' > serial.txt
+  changed_when: false
```
Comment (hcoyote) [inline ansible/roles/demo_certs/tasks/create-ca.yml:49]: is the changed_when something the linter complained about? seems out of place.

File: ansible/roles/demo_certs/tasks/generate-csrs.yml [hunk 2]
```diff
@@ -1,84 +1,78 @@
...
     src: "{{ redpanda_csr_file }}"
     dest: tls/certs/{{ansible_hostname}}/node.csr
-    flat: yes
-
-- name: generate_csrs - Dummy to import role so that we can use its default variables
-  tags:
-    - generate_csrs
-  import_role:
-    name: redpanda_broker
-  when: False
```
Comment (hcoyote) [inline ansible/roles/demo_certs/tasks/generate-csrs.yml:84]: Is this intended to be removed?

File: ansible/roles/demo_certs/tasks/install-certs.yml [hunk 3]
```diff
@@ -1,32 +1,28 @@
...
     - install_certs
-  set_fact:
+  ansible.builtin.set_fact:
     tls: true
-
-- name: install_certs - Apply updated config via reapplying the redpanda_broker role
-  tags: install_certs
-  import_role:
-    name: redpanda_broker
-  when: not skip_node | default(false) | bool
```
Comment (hcoyote) [inline ansible/roles/demo_certs/tasks/install-certs.yml:32]: is this intended tobe removed?

**Review by r-vasquez — APPROVED** [review]
Thanks :white_check_mark: !

File: ansible/roles/redpanda_broker/tasks/install-redpanda.yml [hunk 4]
```diff
@@ -1,9 +1,16 @@
 ---
-- name: add the redpanda repo
-  ansible.builtin.shell: |
-    curl -1sLf {{ redpanda_repo_script }} | sudo -E bash
+- name: Download the Redpanda repo script
+  ansible.builtin.get_url:
+    url: "{{ redpanda_repo_script }}"
+    dest: "/tmp/redpanda_repo_script.sh"
+    mode: '0755'
```
Comment (hcoyote) [inline ansible/roles/redpanda_broker/tasks/install-redpanda.yml:6]: does this end up getting run on every invocation of ansible as well? The previous shell curl|bash thing didn't have a good way of handling only running if changes occurred, so I think it ran every run and updated the repo configs.

aside: we need to do this better, we should be doing curl|bash here as it won't work for airgapped ansible users. Better if we could just define the repo details directly.

File: ansible/roles/demo_certs/tasks/create-ca.yml (hunk 1 above)
Comment (gene-redpanda) [reply ansible/roles/demo_certs/tasks/create-ca.yml:49]: yes, and i strongly considered just turning that linter message off

File: ansible/roles/demo_certs/tasks/generate-csrs.yml (hunk 2 above)
Comment (gene-redpanda) [reply ansible/roles/demo_certs/tasks/generate-csrs.yml:84]: yes, i discuss why in this commit: https://github.com/redpanda-data/deployment-automation/pull/159/commits/520208224ab09aa939394d799fd4bf60a77b2093

File: ansible/roles/demo_certs/tasks/install-certs.yml (hunk 3 above)
Comment (gene-redpanda) [reply ansible/roles/demo_certs/tasks/install-certs.yml:32]: same as above

File: ansible/roles/redpanda_broker/tasks/install-redpanda.yml (hunk 4 above)
Comment (gene-redpanda) [reply ansible/roles/redpanda_broker/tasks/install-redpanda.yml:6]: i think we'd need an airgap toggle to really support that mode, along with a testing environment that is built for airgap. i can't test it right now which means i can't build for it.

File: ansible/roles/demo_certs/tasks/install-certs.yml (hunk 3 above)
Comment (hcoyote) [reply ansible/roles/demo_certs/tasks/install-certs.yml:32]: read through 👍

**Comment by hcoyote** [comment]
I'm +1 on this.

**Pull request merged**